lib/src/script.dart

```diff
@@ -154,18 +154,18 @@ class Script {
   final _outputCloser = StreamCloser<List<int>>();
 
   /// The script's signal handler to terminate the process.
-  late final Future<FutureOr<bool> Function([ProcessSignal])> _signalHandler;
+  late final Future<FutureOr<bool> Function(ProcessSignal)> _signalHandler;
 
   /// Sends a [ProcessSignal] to terminate the process.
   ///
-  /// Defaults to [ProcessSignal.sigterm].
-  ///
-  /// On platforms without signal support, including Windows, the call ignores
-  /// [signal] and the process terminates in a platform specific way.
+  /// If the [Script] is a Unix-style OS process, pass the given signal to the
+  /// process. On platforms without signal support, including Windows, the
+  /// [signal] is ignored and the process terminates in a platform specific way.
   ///
-  /// Returns whether the signal was successfully delivered. Otherwise it
-  /// usually means the process is already dead.
-  FutureOr<bool> signal([ProcessSignal signal = ProcessSignal.sigterm]) async =>
+  /// Returns `true` if the signal is successfully delivered to the process.
+  /// Otherwise the signal could not be sent, usually meaning that the process
+  /// is already dead.
+  Future<bool> kill([ProcessSignal signal = ProcessSignal.sigterm]) async =>
       (await _signalHandler)(signal);
```

Comment:
As I mentioned earlier, this shouldn't invoke the signal handler if the script has already exited.